php/libraries/Candidate.class.inc

```diff
@@ -273,15 +274,20 @@ class Candidate implements \LORIS\StudyEntities\AccessibleResource,
                 EDC_NOT_SPECIFIED
             );
         }
-        if ($dateOfBirth !== null
-            && (DateTime::createFromFormat('Y-m-d', $dateOfBirth) === false
-            || empty($dateOfBirth))
-        ) {
+        if (empty($dateOfBirth)) {
             throw new InvalidArgumentException(
                 "Date of Birth must be specified",
                 DOB_NOT_SPECIFIED
             );
         }
+        $dob = DateTime::createFromFormat('Y-m-d', $dateOfBirth);
```

Comment:
please add `!` in front of the format so that the resulting date has 00:00:00 as its time portion rather than the current time.